core/src/com/unciv/logic/automation/WorkerAutomation.kt

```diff
@@ -281,14 +281,15 @@ class WorkerAutomation(
         val city = tile.getCity()
         if (city == null || city.civInfo != civInfo)
             return false
-        if (tile.improvement != null && !UncivGame.Current.settings.automatedWorkersReplaceImprovements)
+        if (tile.improvement != null && !UncivGame.Current.settings.automatedWorkersReplaceImprovements && UncivGame.Current.gameInfo.currentPlayerCiv.isPlayerCivilization())
```

Comment:
The class has the `civInfo` _owning_ the worker cached - wouldn't that be the safer test?